client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import {
+	get,
+	head,
+	sortBy,
+	toPairs,
+} from 'lodash';
+
+const requestQueue = new Map();
```

Comment:
we could avoid this entirely by adding `retryNumber` to the action object itself in some data-layer meta.  Would be very nice from a debugging perspective to be able to inspect the action and see the attempt number